### DISCUSSION TITLE
PSI隐私求交能力推荐调用方式咨询

### DISCUSSION THREAD

**Discussion by Jerry-zhxf** [secretflow/secretflow#1224] [Q&A]
你好，目前我想调用一下隐语的PSI能力，然后看到一个[教程](https://www.secretflow.org.cn/zh-CN/docs/secretflow/v1.4.0b0/tutorial/PSI_On_SPU)是用SPU去调用PSI进行计算，我认为他应该在[架构图](https://www.yuque.com/secret-flow/admin/mqx1p9#LhTCy)的这个地方：
![image](https://github.com/secretflow/secretflow/assets/36923038/d69a5696-dfc9-427f-84f6-e63d464240f4)
但感觉调用比较麻烦。有看到架构图中上层也有PSI的服务，那上层的PSI是在哪？是Easy-PSI或者PSI呢？
![image](https://github.com/secretflow/secretflow/assets/36923038/e320dad1-ecec-477f-898c-7c0981dd32f6)

**Comment by pchyuan** [comment]
Secretflow有PSI的接口（python），PSI模块也有PSI的接口（C++），可以根据你的需求来选择

**Comment by Chrisdehe** [comment]
SPU调用PSI：https://www.secretflow.org.cn/zh-CN/docs/psi/v0.3.0beta/user_guide/psi_v2
python调用PSI：https://www.secretflow.org.cn/zh-CN/docs/secretflow/v1.5.0b0/developer/benchmark/psi_benchmark
easy PSI平台：https://www.secretflow.org.cn/zh-CN/docs/easy-psi/latest/easy-psi
